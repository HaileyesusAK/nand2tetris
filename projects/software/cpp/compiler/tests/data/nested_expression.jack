((true | false) & (5 > (12 / 6)))
