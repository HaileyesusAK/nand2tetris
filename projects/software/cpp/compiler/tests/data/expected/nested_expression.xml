<expression>
    <term>
        <symbol> ( </symbol>
        <expression>
            <term>
                <symbol> ( </symbol>
                <expression>
                    <term>
                        <keyword> true </keyword>
                    </term>
                    <symbol> | </symbol>
                    <term>
                        <keyword> false </keyword>
                    </term>
                </expression>
                <symbol> ) </symbol>
            </term>
            <symbol> &amp; </symbol>
            <term>
                <symbol> ( </symbol>
                <expression>
                    <term>
                        <integerConstant> 5 </integerConstant>
                    </term>
                    <symbol> &gt; </symbol>
                    <term>
                        <symbol> ( </symbol>
                        <expression>
                            <term>
                                <integerConstant> 12 </integerConstant>
                            </term>
                            <symbol> / </symbol>
                            <term>
                                <integerConstant> 6 </integerConstant>
                            </term>
                        </expression>
                        <symbol> ) </symbol>
                    </term>
                </expression>
                <symbol> ) </symbol>
            </term>
        </expression>
        <symbol> ) </symbol>
    </term>
</expression>
